### ISSUE TITLE
Obtaining logits during model.generate()

### ISSUE THREAD

**Issue by QiyaoWei** [huggingface/trl#1425]
What I am trying to do is explained in the title. As far as I understand, there are two ways of doing this

(1) pass in the output_logits=True parameter into model.generate() directly
(2) forward pass the model on the generated textual output

MWE here, is there something that I am not understanding correctly?

```
import random
random.seed("1234")
from transformers import AutoTokenizer, AutoModelForCausalLM, set_seed, pipeline
set_seed(1234)

model_name="lvwerra/gpt2-imdb"
model = AutoModelForCausalLM.from_pretrained(model_name).to("cuda")
tokenizer = AutoTokenizer.from_pretrained(model_name)
tokenizer.pad_token = tokenizer.eos_token

# I want to have the logits when I do generation
encoding = tokenizer(["Hi there, how are you?"], return_tensors="pt").to("cuda")
generation_output = model.generate(**encoding, return_dict_in_generate=True, output_logits=True)
# To be clear, this method returns the logits of the generated tokens, which does not include the prompt, but that is a minor detail

# obtain textual and logits output from gen
sequences = generation_output.sequences
sanity_check_logits = generation_output.logits

# we already have our encoding, now we can look at the logits from our models
dummy_var = {"input_ids": sequences}
model_output = model(**dummy_var)

# assert sanity_check_logits == model_output.logits fails
```

**Comment by younesbelkada** [comment]
Hi @QiyaoWei 
Yes this should be correct imo, what is exactly the issue you are facing?

**Comment by QiyaoWei** [comment]
Hi @younesbelkada Problem solved here https://github.com/huggingface/transformers/issues/29664
